packages/core/src/account/base.ts

```diff
@@ -30,11 +31,28 @@ export enum DeploymentState {
 export interface BaseSmartAccountParams<
   TTransport extends SupportedTransports = Transport
 > {
+  /* Required */
   rpcClient: string | PublicErc4337Client<TTransport>;
-  entryPointAddress: Address;
   factoryAddress: Address;
-  owner?: SmartAccountSigner | undefined;
   chain: Chain;
+
+  /* Optional */
+
+  /**
+   * The address of the entry point contract.
+   * If not provided, the default entry point contract will be used.
+   * Check out https://docs.alchemy.com/reference/eth-supportedentrypoints for all the supported entrypoints
+   */
+  entryPointAddress?: Address;
```

Comment:
same question as on provider (though here you moved closer to my suggested approach):

can't we have this take in a default value of getDefaultEntryPointContract(params.chain);

very loose prototype:

```
class BaseAccount {
  constructor({
    chain,
    entryPointAddress = getDefaultEntryPointAddress(chain),
  }: {
    chain: Chain;
    entryPointAddress?: Address;
  }) {
    console.log(entryPointAddress);
  }
}
```

^this type checks for me and I can access the entryPoint value directly as a non-optional type:

![image](https://github.com/alchemyplatform/aa-sdk/assets/43521356/332eca7b-da5c-47ff-8740-742c1dbb6c88)